indico/modules/auth/controllers.py

```diff
@@ -772,12 +772,23 @@ def _request_token(self):
             # username than the one he expects. But he still gets back into his profile.
             # Showing a list of usernames would be a little bit more user-friendly but less
             # secure as we'd expose valid usernames for a specific user to an untrusted person.
-            identity = next(iter(user.local_identities))
-            _send_confirmation(form.email.data, 'reset-password', '.resetpass', 'auth/emails/reset_password.txt',
-                               {'user': user, 'username': identity.identifier},
-                               data={'id': identity.id, 'hash': crc32(identity.password_hash)})
+            if user is None:
+                # TODO: Send email informing that the email is not associated to any user?
+                #       We could add in that email a link to create a new account.
+                logger.info('Password reset requested for non-existing user %s', user)
+            elif not user.local_identities:
+                # XXX: Should we allow creating a new identity instead? Would be user-friendly for sure!
+                # TODO: Send email informing that the user has no local account?
+                #       We could add in that email a link to create a new local identity.
```

Comment:
I would go a step further and show a message on the password reset page that a new local login will be created - and maybe show a list of existing alternative login methods if there are any, or at least the most recently used one?

Reply:
This will be even more useful when combined with the PR that allows the email to login - in that case we don't even need to prompt for a user name.

In any case, being able to attach a local identity to an account which does not have one via "reset password" is SUPER useful and actually something we'll need for Indico Global! (I was actually about to put that on my own todo list, but maybe it fits nicely in here ;))